### PR TITLE
add centralize slug reference

### PR DESCRIPTION
**Pull request by brettski**
Community slugs now part of centralized slug collection
Slug written on create
changeSlug mutation added

### CODE REVIEW

File: src/dataSources/cloudFirestore/community.js [hunk 1]
```diff
@@ -167,11 +222,67 @@ const community = dbInstance => {
...
+        'invalid communityId provided, unable to change community slug',
+      );
+    const cleanCommunity = scrubCommunity({
+      community: {
+        slug: newSlug,
+      },
+      user,
+    });
+    const currentSlug = docSnapshot.get('slug');
+    const currentSlugDocRef = slugStore(dbInstance).getSlugDocRef(currentSlug);
```
Comment (csell5) [inline src/dataSources/cloudFirestore/community.js:253]: any reason you don't batch these three into a promise.all?

File: src/dataSources/cloudFirestore/community.js [hunk 2]
```diff
@@ -119,24 +125,73 @@ const community = dbInstance => {
...
+    const newSlug = cleanCommunity.slug;
+    const slugInUse = await isSlugTaken(newSlug);
+    if (slugInUse)
+      throw new Error(
+        'Slug in use, it cannot be used to create a new community',
+      );
+
+    const communityDocRef = communityCol.doc(); // creates a random id
+    dlog('new community id %s');
+    const slugDocRef = slugStore(dbInstance).getSlugDocRef(newSlug);
```
Comment (csell5) [inline src/dataSources/cloudFirestore/community.js:157]: same here with batching these calls

File: src/dataSources/cloudFirestore/community.js (hunk 1 above)
Comment (brettski) [reply src/dataSources/cloudFirestore/community.js:253]: getSlugRef doesn't return a Promise. It is a reference created by the sdk. This reference is needed for the batch write to Firestore

File: src/dataSources/cloudFirestore/community.js (hunk 2 above)
Comment (brettski) [reply src/dataSources/cloudFirestore/community.js:157]: getSlugRef doesn't return a Promise. It is a reference created by the sdk. This reference is needed for the batch write to Firestore

**Review by csell5 — APPROVED** [review]

**Pull request merged**